### PR TITLE
Distortion Audio Effect

### PR DESCRIPTION
**Pull request by pdoogs**
Just like the Echo and Chorus effect.

### CODE REVIEW

File: src/audio/win/audio_device.cpp [hunk 1]
```diff
@@ -355,6 +355,63 @@ struct AudioDeviceImpl LUMIX_FINAL : public AudioDevice
...
+
+		DWORD buffer_status;
+		if (FAILED(buffer.handle->GetStatus(&buffer_status))) return;
+
+		buffer.handle->Stop();
+		DWORD res = 0;
+
+		IDirectSoundFXDistortion8* distortion = NULL;
+		if (FAILED(buffer.handle8->GetObjectInPath(
+			GUID_DSFX_STANDARD_CHORUS, 0, IID_IDirectSoundFXDistortion8, (LPVOID*)&distortion)))
```
Comment (nem0) [inline src/audio/win/audio_device.cpp]: chorus ->distortion

File: src/audio/audio_device.h [hunk 2]
```diff
@@ -49,6 +49,19 @@ class LUMIX_AUDIO_API AudioDevice
...
 		float right_delay) = 0;
+	virtual void setChorus(BufferHandle handle,
+		float wet_dry_mix,
+		float depth,
+		float feedback,
+		float frequency,
+		float delay,
+		i32 phase) = 0;
+	virtual void setDistortion(BufferHandle handle,
+		float distortionEdge,
```
Comment (nem0) [inline src/audio/audio_device.h]: distortionEdge -> distortion_edge

File: src/audio/audio_scene.cpp [hunk 3]
```diff
@@ -40,6 +41,28 @@ struct EchoZone
...
+	float depth;
+	float feedback;
+	float frequency;
+	i32 phase;
+};
+
+struct DistortionZone
+{
+	Entity entity;
+	float radius;
```
Comment (nem0) [inline src/audio/audio_scene.cpp]: underscore_case

**Comment by pdoogs** [comment]
I'll implement the fixes and only issue another pull request once the plugins are in.

**Comment by nem0** [comment]
Closing this for now

**Pull request closed without merging**